pkg/middleware/validate_action_url_test.go

```diff
@@ -0,0 +1,148 @@
+package middleware
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMiddlewareValidateActionUrl(t *testing.T) {
+	tests := []struct {
+		name                string
+		method              string
+		path                string
+		actionsAllowPostURL string
+		addHeader           bool
+		code                int
+	}{
+		{
+			name:                "POST without action header",
+			method:              "POST",
+			path:                "/api/plugins/org-generic-app",
+			actionsAllowPostURL: "",
+			addHeader:           false,
+			code:                http.StatusOK,
+		},
+		{
+			name:                "POST with action header, no paths defined",
+			method:              "POST",
+			path:                "/api/plugins/org-generic-app",
+			actionsAllowPostURL: "",
+			addHeader:           true,
+			code:                http.StatusMethodNotAllowed,
+		},
+		{
+			name:                "POST action with allowed path",
+			method:              "POST",
+			path:                "/api/plugins/org-generic-app",
+			actionsAllowPostURL: "/api/plugins/*",
+			addHeader:           true,
+			code:                http.StatusOK,
+		},
+		{
+			name:                "POST action without invalid path",
+			method:              "POST",
+			path:                "/api/notplugins/org-generic-app",
+			actionsAllowPostURL: "/api/plugins/*",
+			addHeader:           true,
+			code:                http.StatusMethodNotAllowed,
+		},
```

Comment:
Should add cases for PUT and for disallowed verbs (maybe start with GET, DELETE)

Reply:
working on it, definitely want to get full coverage and expected behavior

Reply:
@DanCech ptal, i think they are covered now with expected results